doc/sphinx/src/fortran.rst

```diff
@@ -0,0 +1,693 @@
+       allocate(press(ncell), pmax(ncell), vol(ncell), spvol(ncell))
+       allocate(sie(ncell), temp(ncell), bmod(ncell), dpde(ncell), cv(ncell))
+       allocate(frac_mass(ncell, nmat), frac_vol(ncell, nmat), frac_sie(ncell, nmat))
+   
+       ! Initialize arrays
+       eos_offsets = 1
+       offsets = [(i, i=1,ncell)]
+       press = 0.0d0
+       pmax = 0.0d0
+       vol = 0.0d0
+       spvol = 0.0d0
+       sie = 0.0d0
+       bmod = 0.0d0
+       dpde = 0.0d0
+       cv = 0.0d0
+       frac_mass = 1.0d0
+       frac_vol = 1.0d0
+       frac_sie = 1.0d0
+       
+       ! Set up density-temperature grid
+       idx = 0
+       do i = 1, ndens
+           do j = 1, ntemp
+               idx = idx + 1
+               ! Logarithmic spacing
+               density(idx) = dens_min * (dens_max/dens_min)**((i-1.0d0)/(ndens-1.0d0))
+               spvol(idx) = 1.0d0 / density(idx)
+               temp(idx) = temp_min * (temp_max/temp_min)**((j-1.0d0)/(ntemp-1.0d0))
+           end do
+       end do
+   
+       ! Initialize EOS
+       res = init_sg_eos_f(nmat, eos)
+       if (res /= 0) then
+           print *, "Error initializing EOS:", res
+           stop
+       end if
+   
+       ! Set EOSPAC options (all zeros for default behavior)
+       eospac_opts = 0.0d0
+   
+       ! Initialize material with EOSPAC
+       mat = 1
+       matid = 9999  ! Example material ID
+       res = init_sg_eospac_f(mat, eos, matid, eospac_opts)
+       if (res /= 0) then
+           print *, "Error initializing EOSPAC:", res
+           stop
+       end if
+   
+       ! Calculate EOS properties
+       ! Note: Temperature is in eV for get_sg_eos_f
+       res = get_sg_eos_f(nmat, ncell, cell_dim, option, &
+                         eos_offsets, eos, offsets, &
+                         press, pmax, vol, spvol, sie, &
+                         temp, bmod, dpde, cv, &
+                         frac_mass, frac_vol, frac_sie)
+       if (res /= 0) then
+           print *, "Error in get_sg_eos_f:", res
+           stop
+       end if
+   
+       ! Print some results
+       print *, "Results for first grid point:"
+       print *, "Density = ", density(1), " g/cm^3"
+       print *, "Temperature = ", temp(1), " eV"
+       print *, "Pressure = ", press(1), " dyne/cm^2"
+       print *, "SIE = ", sie(1), " erg/g"
+   
+       ! Clean up
+       deallocate(eos_offsets, offsets, press, pmax, vol, spvol)
+       deallocate(sie, temp, bmod, dpde, cv, density)
+       deallocate(frac_mass, frac_vol, frac_sie)
+   
+       res = finalize_sg_eos_f(nmat, eos)
+       if (res /= 0) then
+           print *, "Error finalizing EOS:", res
+           stop
+       end if
+   
+   end program tabulated_eos_example
+
+.. note::
+   
+   When using the ``get_sg_eos_f`` function, temperature inputs should be provided in electron volts (eV), not Kelvin. This is different from some other functions in the interface that expect temperature in Kelvin.
```

Comment:
This is an example of something we might want to change down the line. :)

Reply:
This is actually code that maybe shouldn't live in `singularity-eos` to be honest since it's very specific.

I'd advocate that we should expose a very general Fortran interface, but that the `get_sg_eos` interface is waaaaay too specific. There's an advantage to having it in our code since we test its capabilities, but I think we should try to capture those some other way and/or explore other ways of testing the library against this interface.

Reply:
Yeah absolutely. That said, if there is a desire to access PTE on the fortran side, we should think how we'd like the generic interface to look.

Reply:
True enough! I think we can discuss future possibilities now that we are going to have three different Fortran codes interfacing